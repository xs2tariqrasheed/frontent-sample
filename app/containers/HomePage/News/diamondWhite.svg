<?xml version="1.0" encoding="UTF-8"?>
<svg width="157px" height="232px" viewBox="0 0 157 232" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: Sketch 64 (93537) - https://sketch.com -->
    <title>diamondWhite</title>
    <desc>Created with Sketch.</desc>
    <g id="homepage-update" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <g id="diamondWhite">
            <rect id="Rectangle-Copy-17" fill="#E7E6FF" x="0" y="0" width="157" height="232"></rect>
            <g id="components/graphics/diamond-white">
                <polygon id="Rectangle" fill="#FFFFFF" points="78 0 157 117 78 232 2.84217094e-13 117"></polygon>
                <path d="M115.202865,169.278808 L125.82753,185.030522 L124.998492,185.589715 L114.373827,169.838001 L115.202865,169.278808 Z M41.8815321,168.638943 L42.7006841,169.212519 L31.8027318,184.776408 L30.9835798,184.202831 L41.8815321,168.638943 Z M103.459814,151.869019 L114.084479,167.620733 L113.255441,168.179926 L102.630776,152.428212 L103.459814,151.869019 Z M53.9266372,151.43675 L54.7457893,152.010326 L43.847837,167.574215 L43.0286849,167.000639 L53.9266372,151.43675 Z M91.7167627,134.45923 L102.341428,150.210944 L101.51239,150.770137 L90.8877251,135.018423 L91.7167627,134.45923 Z M65.9717424,134.234557 L66.7908944,134.808133 L55.8929421,150.372022 L55.0737901,149.798446 L65.9717424,134.234557 Z M79.9737117,117.049441 L90.5983769,132.801155 L89.7693393,133.360348 L79.1446741,117.608634 L79.9737117,117.049441 Z M78.0168475,117.032364 L78.8359996,117.60594 L67.9380473,133.169829 L67.1188953,132.596253 L78.0168475,117.032364 Z M90.0619527,99.8301709 L90.8811047,100.403747 L79.9831525,115.967636 L79.1640004,115.39406 L90.0619527,99.8301709 Z M68.2306607,99.6396521 L78.8553259,115.391366 L78.0262883,115.950559 L67.4016231,100.198845 L68.2306607,99.6396521 Z M102.107058,82.627978 L102.92621,83.2015544 L92.0282576,98.7654432 L91.2091056,98.1918668 L102.107058,82.627978 Z M56.4876097,82.2298631 L67.1122749,97.981577 L66.2832373,98.5407699 L55.6585722,82.789056 L56.4876097,82.2298631 Z M114.152163,65.425785 L114.971315,65.9993615 L104.073363,81.5632503 L103.254211,80.9896739 L114.152163,65.425785 Z M44.7445588,64.820074 L55.3692239,80.5717879 L54.5401864,81.1309808 L43.9155212,65.379267 L44.7445588,64.820074 Z M126.197268,48.2235921 L127.01642,48.7971685 L116.118468,64.3610574 L115.299316,63.7874809 L126.197268,48.2235921 Z M33.0015078,47.410285 L43.626173,63.1619989 L42.7971354,63.7211918 L32.1724702,47.9694779 L33.0015078,47.410285 Z" id="Shape" fill="#E7E6FF"></path>
            </g>
            <rect id="Rectangle-Copy-17" fill="#F3F2FF" opacity="0.5" x="0" y="0" width="157" height="232"></rect>
        </g>
    </g>
</svg>