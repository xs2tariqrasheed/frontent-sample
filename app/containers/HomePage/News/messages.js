/*
 * NewsCard Messages
 *
 * This contains all the text for the NewsCard component.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.components.News';

export default defineMessages({
  sectionTitle: {
    id: `${scope}.sectionTitle`,
    defaultMessage: 'News & Updates',
  },
  updates: {
    oneTitle: {
      id: `${scope}.updates.oneTitle`,
      defaultMessage: 'How to choose your golfer',
    },
    oneLabel: {
      id: `${scope}.updates.oneLabel`,
      defaultMessage: 'Nonfungible.com gives tips on choosing golfers',
    },
    oneMessage: {
      id: `${scope}.updates.oneMessage`,
      defaultMessage:
        'We have identified what we believe to be the best strategy to date to build a great team of golfers.',
    },
    twoTitle: {
      id: `${scope}.updates.twoTitle`,
      defaultMessage: 'Check out who we are and what we stand for.',
    },
    twoLabel: {
      id: `${scope}.updates.twoLabel`,
      defaultMessage: 'A Deep Dive into Blocklete Games™',
    },
    twoMessage: {
      id: `${scope}.updates.twoMessage`,
      defaultMessage:
        'Blockchain technology is at the core of all of our game and tech solutions because of what it enables — a player-driven game economy.',
    },
    threeTitle: {
      id: `${scope}.updates.threeTitle`,
      defaultMessage: 'Our first 500 minted golfers have their own tournament!',
    },
    threeLabel: {
      id: `${scope}.updates.threeLabel`,
      defaultMessage: 'First Tournament Announcement',
    },
    threeMessage: {
      id: `${scope}.updates.threeMessage`,
      defaultMessage:
        'Only golfer tokens 1–500 are able to enter this event. The tournament prize pot is set at $1,000!',
    },
  },
});
