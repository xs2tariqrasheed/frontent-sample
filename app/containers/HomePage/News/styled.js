import styled from 'styled-components';
import mq from '../../../styles/mq';
import Img from '../../../components/Img';
const AboutBkg = 'https://www.blockletegames.com/images/AboutBkg.png';

export const Container = styled.div`
  background: url(${AboutBkg}) repeat;
`;

export const ImageContainer = styled(Img)`
  margin-right: 30px;
  margin-left: auto;

  ${mq.sm} {
    margin-right: 0px;
    max-width: 100%;
    margin-top: 20px;
  }
`;

export const Button = styled.button`
  cursor: pointer;
  width: 234px;
  height: 50px;
  object-fit: contain;
  font-family: Roboto;
  font-size: 16px;
  font-weight: bold;
  font-stretch: normal;
  font-style: normal;
  line-height: normal;
  letter-spacing: 1px;
  text-align: center;
  color: #ffffff;
  background-color: #3b38c6;
  border: none;
  border-radius: 4px;
  transition: background-color 1s ease;

  &:hover {
    background-color: #201f4b;
  }

  ${mq.sm} {
    display: block;
    margin-top: 115px;
    margin-bottom: 15px;
  }
`;
