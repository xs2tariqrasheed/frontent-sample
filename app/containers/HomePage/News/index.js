/*
 * About
 *
 * This is the About block on the front page.
 *
 */
import React from 'react';
import { FormattedMessage } from 'react-intl';
import { Container } from './styled';
import NewsCard from '../../../components/NewsCard';
import messages from './messages';
const founders = 'https://www.blockletegames.com/images/founders.png';
const article1 = 'https://www.blockletegames.com/images/article1.png';
const deep = 'https://www.blockletegames.com/images/deep-dive.png';

export default function News() {
  return (
    <Container>
      <section className="container mx-auto px-5 pt-12 pb-24">
        <FormattedMessage {...messages.sectionTitle}>
          {txt => (
            <h1 className="text-blueberry callout-underline mb-12 text-4xl tracking-wide font-heavy uppercase">
              {txt}
            </h1>
          )}
        </FormattedMessage>
        <div className="sm:grid grid-cols-3 grid-flow-col gap-6">
          <NewsCard
            picture={article1}
            link="https://nonfungible.com/blog/arena-golf-how-to-choose-your-golfers"
            label={messages.updates.oneLabel}
            title={messages.updates.oneTitle}
            message={messages.updates.oneMessage}
          />
          <NewsCard
            picture={deep}
            link="https://medium.com/in-the-arena-sports/a-deep-dive-into-the-world-of-arena-golf-36d3282df48a"
            label={messages.updates.twoLabel}
            title={messages.updates.twoTitle}
            message={messages.updates.twoMessage}
          />
          <NewsCard
            picture={founders}
            link="https://medium.com/in-the-arena-sports/arena-golfs-first-tournament-announcement-cc67e14e42bb"
            label={messages.updates.threeLabel}
            title={messages.updates.threeTitle}
            message={messages.updates.threeMessage}
          />
        </div>
      </section>
    </Container>
  );
}
