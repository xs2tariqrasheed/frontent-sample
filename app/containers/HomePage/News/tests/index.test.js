import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory, BrowserRouter } from 'react-router-dom';
import configureStore from '../../../../configureStore';

import News from '../index';

describe('<About />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store.store}>
        <BrowserRouter>
          <IntlProvider locale="en">
            <News />
          </IntlProvider>
        </BrowserRouter>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
