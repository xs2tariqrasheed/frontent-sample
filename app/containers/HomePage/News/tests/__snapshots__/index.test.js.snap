// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<About /> should render and match the snapshot 1`] = `
<div
  class="styled__Container-og6jsg-0 kZbrPR"
>
  <section
    class="container mx-auto px-5 pt-12 pb-24"
  >
    <h1
      class="text-blueberry callout-underline mb-12 text-4xl tracking-wide font-heavy uppercase"
    >
      News & Updates
    </h1>
    <div
      class="sm:grid grid-cols-3 grid-flow-col gap-6"
    >
      <article
        class="bg-white rounded overflow-hidden shadow mb-5 sm:mb-0"
      >
        <a
          class="no-underline"
          href="https://nonfungible.com/blog/arena-golf-how-to-choose-your-golfers"
          target="_blank"
        >
          <img
            alt=""
            class="w-full"
            src="IMAGE_MOCK"
          />
          <h2
            class="ribbon-cap tracking-wide pl-6 py-1 pr-2 bg-yellow-itas-dark-gold text-itas-dark-purple uppercase text-xs font-heavy"
          >
            Nonfungible.com gives tips on choosing golfers
          </h2>
          <div
            class="px-6 py-4"
          >
            <h1
              class="font-bold mb-2 text-sm text-itas-lavender"
            >
              How to choose your golfer
            </h1>
            <p
              class="text-sm text-black"
            >
              We have identified what we believe to be the best strategy to date to build a great team of golfers.
            </p>
          </div>
        </a>
      </article>
      <article
        class="bg-white rounded overflow-hidden shadow mb-5 sm:mb-0"
      >
        <a
          class="no-underline"
          href="https://medium.com/in-the-arena-sports/a-deep-dive-into-the-world-of-arena-golf-36d3282df48a"
          target="_blank"
        >
          <img
            alt=""
            class="w-full"
            src="IMAGE_MOCK"
          />
          <h2
            class="ribbon-cap tracking-wide pl-6 py-1 pr-2 bg-yellow-itas-dark-gold text-itas-dark-purple uppercase text-xs font-heavy"
          >
            A Deep Dive into Blocklete Games™
          </h2>
          <div
            class="px-6 py-4"
          >
            <h1
              class="font-bold mb-2 text-sm text-itas-lavender"
            >
              Check out who we are and what we stand for.
            </h1>
            <p
              class="text-sm text-black"
            >
              Blockchain technology is at the core of all of our game and tech solutions because of what it enables — a player-driven game economy.
            </p>
          </div>
        </a>
      </article>
      <article
        class="bg-white rounded overflow-hidden shadow mb-5 sm:mb-0"
      >
        <a
          class="no-underline"
          href="https://medium.com/in-the-arena-sports/arena-golfs-first-tournament-announcement-cc67e14e42bb"
          target="_blank"
        >
          <img
            alt=""
            class="w-full"
            src="IMAGE_MOCK"
          />
          <h2
            class="ribbon-cap tracking-wide pl-6 py-1 pr-2 bg-yellow-itas-dark-gold text-itas-dark-purple uppercase text-xs font-heavy"
          >
            First Tournament Announcement
          </h2>
          <div
            class="px-6 py-4"
          >
            <h1
              class="font-bold mb-2 text-sm text-itas-lavender"
            >
              Our first 500 minted golfers have their own tournament!
            </h1>
            <p
              class="text-sm text-black"
            >
              Only golfer tokens 1–500 are able to enter this event. The tournament prize pot is set at $1,000!
            </p>
          </div>
        </a>
      </article>
    </div>
  </section>
</div>
`;
